service/awsconfig/v5/resource/cloudformation/current.go

```diff
@@ -16,103 +18,129 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
-	}
-	workerCloudConfigVersion, err := getStackOutputValue(outputs, workerCloudConfigVersionOutputKey)
-	if err != nil {
-		return StackState{}, microerror.Mask(err)
-	}
-	versionBundleVersion, err := getStackOutputValue(outputs, versionBundleVersionOutputKey)
-	if IsNotFound(err) {
-		// Since we are transitioning between versions we will have situations in
-		// which old clusters are updated to new versions and miss the version
-		// bundle version in the CF stack outputs. We ignore this problem for now
-		// and move on regardless. The reconciliation will detect the guest cluster
-		// needs to be updated and once this is done, we should be fine again.
-		//
-		// TODO remove this condition as soon as all guest clusters in existence
-		// obtain a version bundle version.
-		versionBundleVersion = ""
-	} else if err != nil {
-		return StackState{}, microerror.Mask(err)
+		return StackState{}, nil
 	}
 
-	outputStackState := StackState{
-		Name: stackName,
+	// Here we finally construct the current state.
+	currentState := StackState{}
+	{
+		outputs := describeOutput.Stacks[0].Outputs
 
-		MasterImageID:            masterImageID,
-		MasterInstanceType:       masterInstanceType,
-		MasterCloudConfigVersion: masterCloudConfigVersion,
+		masterImageID, err := getStackOutputValue(outputs, masterImageIDOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		masterInstanceType, err := getStackOutputValue(outputs, masterInstanceTypeOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		masterCloudConfigVersion, err := getStackOutputValue(outputs, masterCloudConfigVersionOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		workers, err := getStackOutputValue(outputs, workersOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		workerImageID, err := getStackOutputValue(outputs, workerImageIDOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		workerInstanceType, err := getStackOutputValue(outputs, workerInstanceTypeOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		workerCloudConfigVersion, err := getStackOutputValue(outputs, workerCloudConfigVersionOutputKey)
+		if err != nil {
+			return StackState{}, microerror.Mask(err)
+		}
+		versionBundleVersion, err := getStackOutputValue(outputs, versionBundleVersionOutputKey)
+		if IsNotFound(err) {
+			// Since we are transitioning between versions we will have situations in
+			// which old clusters are updated to new versions and miss the version
+			// bundle version in the CF stack outputs. We ignore this problem for now
+			// and move on regardless. The reconciliation will detect the guest cluster
+			// needs to be updated and once this is done, we should be fine again.
+			//
+			// TODO remove this condition as soon as all guest clusters in existence
+			// obtain a version bundle version.
```

Comment:
This comment confuses me. Guest clusters without a version bundle version are routed to the v1 legacy resource set. 

I think the case here is all clusters need a cloud formation stack output with the version bundle version.

Reply:
This is for transition. At first you have a guest cluster with version "1" which does not support the version bundle version in the stack outputs. Then we have the guest clusters with version "2" which support it. When we go and change the version bundle version in the CR we will reconcile here and see that the CF stack does not have the version bundle version and we would fail. So we handle this specific case for the transition period and be good. The rest is then sorted by the reconciliation down the road where we compare current and desired state. For the comparison it does not matter if the version bundle version is "Lutz" an empty string or "1.0.0". As soon as it is different to the desired state we update.

Reply:
Ack. That makes sense.

Reply:
Do you miss something in the comment we have in the code? Can I improve it somehow?

Reply:
My confusion was the resource routing issue I mentioned. But the condition applies to the stack output and this code is getting the output. So I think the comment is clear as it is.